source/contracts/trading/CancelOrder.sol

```diff
@@ -56,15 +59,12 @@ contract CancelOrder is CashAutoConverter, Extractable, ReentrancyGuard, ICancel
             if (_type == Order.Types.Bid) {
                 for (uint8 _i = 0; _i < _market.getNumberOfOutcomes(); ++_i) {
                     if (_i != _outcome) {
-                        _market.getShareToken(_i).transfer(_sender, _sharesEscrowed);
+                        _market.getShareToken(_i).trustedCancelOrderTransfer(_market, _sender, _sharesEscrowed);
```

Comment:
I'm assuming `_market` being unvalidated here is on the TODO list for later?  Should I assume the same for all cases of trusted inputs throughout this PR, or should I call them all out to be sure they don't slip through the cracks?

Reply:
Yea I'm doing all of that in another PR currently.

Reply:
Which i intend to sit on for a bit until we can get auditor feedback on the two options